asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/flow_utils.py

```diff
@@ -0,0 +1,267 @@
+        )
+
+
+def get_mse_category_data(category: str, collection_date: str) -> pd.DataFrame:
+    """
+    Gets data from a specific Money Saving Expert category, collected on a certain date.
+
+    Args:
+        category (str): An MSE category
+        collection_date (str): A date in the format "YYYY_MM_DD"
+    Returns:
+        pd.DataFramedataframe with most up to date category/sub-forum data
+    """
+    if category == "energy":
+        return load_s3_data(
+            bucket_name=S3_BUCKET,
+            file_path=f"data/mse/outputs/mse_data_category_energy.parquet",
+        )
+
+    return load_s3_data(
+        bucket_name=S3_BUCKET,
+        file_path=f"data/mse/outputs/mse_data_category_{category}_{collection_date}.parquet",
+    )
+
+
+def get_all_mse_data(collection_date: str) -> pd.DataFrame:
+    """
+    Gets data from all Money Saving Expert categories.
+
+    Args:
+        collection_date (str): A date in the format "YYYY_MM_DD"
+    Returns:
+        pd.DataFrame: a dataframe with data from all categories
+    """
+    categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+    all_mse_data = pd.DataFrame()
+    for cat in categories:
+        if (
+            cat != "energy"
+        ):  # this is a temporary fix, while we sort the collection for this category
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_{cat}_{collection_date}.parquet",
+            )
+        else:
+            aux = load_s3_data(
+                bucket_name=S3_BUCKET,
+                file_path=f"data/mse/outputs/mse_data_category_energy.parquet",
+            )
+        all_mse_data = pd.concat([all_mse_data, aux])
+
+    return all_mse_data.drop_duplicates().reset_index(drop=True)
+
+
+def get_mse_data(category: str, collection_datetime: str) -> pd.DataFrame:
+    """
+    Get MSE data for a specific category and collection date.
+    Existing categories are: "green-ethical-moneysaving", "lpg-heating-oil-solid-other-fuels", "energy", "is-this-quote-fair"
+    Additionally, you can also use "sample" (to get the initial sample collected) or "all" (to get all the MSE data collected).
+
+    Args:
+        category (str): An MSE category, "sample" or "all"
+        collection_datetime (str): A date in the format "YYYY_MM_DD"
+
+    Returns:
+        pd.DataDrame: a dataframe with the MSE data
+    """
+    mse_categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+
+    accepted_categories = mse_categories + ["all"]
+    try:
+        # Check if the category is in the list if one of the accepted categories
+        category_index = accepted_categories.index(category)
```

Comment:
is the variable "category_index" needed or could you just have "accepted_categories.index(category)"